File: funciones.c
#include <stdio.h>

float calcBmi(float h,float w);

float pIdeal(float bmi, int altura);

int main(){
  
    int users;
    int b = 0;
    int n = 0;
    int s = 0;
    int o = 0;

    printf("Ingrese la cantidad de personas: ");
    scanf("%d",&users);

    float height[users];
    float weight[users];
    float bmi[users];
    float Peso_IDeal[users];
    
    float porcBaj, porcNor, porcSob, porcOb;
    float BMI_Normal = 24.9;
   
    for(int i=0;i<users;i++){
        printf("\nUsuario: %d",i+1);

        printf("\nIngrese su altura en cms: ");
        scanf("%f",&height[i]);

        printf("Ingrece su peso en kg: ");
        scanf("%f",&weight[i]);

        bmi[i] = calcBmi(height[i],weight[i]);
        printf("\nSu indice de masa corporal es:%.2f kg/cm^2",bmi[i]);
            
    if(bmi[i]<=18.5){
        printf("\nBajo peso\n");
        Peso_IDeal[i] = pIdeal(BMI_Normal, height[i]);
        printf("\nTu peso ideal es: %.2fkg", Peso_IDeal[i]);
        b++;    
        }
    else if (bmi[i]>18.5 && bmi[i]<=24.9){
        printf("\nPeso normal\n");
        n++;
    }
    else if (bmi[i]>24.9 && bmi[i]<29.9){
        printf("\nSobrepeso\n");
        Peso_IDeal[i] = pIdeal(BMI_Normal, height[i]);
        printf("\nTu peso ideal es: %.2fkg", Peso_IDeal[i]);
        s++;
    }
    else if(bmi[i]>=29.9){
        printf("\nObesidad\n");
        Peso_IDeal[i] = pIdeal(BMI_Normal, height[i]);
        printf("\nTu peso ideal es: %.2fkg", Peso_IDeal[i]);
        o++;
    }
    }

    porcBaj = (100*b)/users;
    porcNor = (100*n)/users;
    porcSob = (100*s)/users;
    porcOb = (100*o)/users;
    printf("\nPorcentaje de personas en Bajo peso: %.1f%%", porcBaj);
    printf("\nPorcentaje de personas en Peso Normal: %.1f%%", porcNor);
    printf("\nPorcentaje de personas en Sobrepeso: %.1f%%", porcSob);
    printf("\nPorcentaje de personas en Obesidad: %.1f%%", porcOb);

}

float calcBmi(float h,float w){
    float bmi;
    
    
    bmi=w/((h/100.0)*(h/100.0));
    return bmi;
}

float pIdeal(float bmi, int altura){
    float altura_1 = altura / 100.0;
    float peso = (bmi)*(altura_1*altura_1);

    return peso;
}

File: funciones.py
def calc_bmi(height, weight):
    bmi = weight / ((height / 100.0) ** 2)
    return bmi

def pIdeal(bmi, altura):
    altura_1 = altura / 100.0
    peso = bmi * (altura_1 ** 2)
    return peso

def main():
    users = int(input("Ingrese la cantidad de personas: "))
    
    height = []
    weight = []
    bmi = []
    peso_ideal = []
    
    b = n = s = o = 0
    BMI_NORMAL = 24.9
    
    for i in range(users):
        print(f"\nUsuario: {i + 1}")
        
        h = float(input("Ingrese su altura en cms: "))
        height.append(h)
        
        w = float(input("Ingrese su peso en kg: "))
        weight.append(w)
        
        bmis = calc_bmi(h, w)
        bmi.append(bmis)
        print(f"\nSu índice de masa corporal es: {bmis:.2f} kg/m^2")
        
        if bmis <= 18.5:
            print("\nBajo peso")
            ideal_weight = pIdeal(BMI_NORMAL, h)
            peso_ideal.append(ideal_weight)
            print(f"\nTu peso ideal es: {ideal_weight:.2f} kg")
            b += 1
        elif bmis>18.5 and bmis<=24.9:
            print("Peso normal")
            n += 1
        elif bmis>24.9 and bmis<29.9:
            print("Sobrepeso")
            ideal_weight = pIdeal(BMI_NORMAL, h)
            peso_ideal.append(ideal_weight)
            print(f"\nTu peso ideal es: {ideal_weight:.2f} kg")
            s += 1
        elif bmis>=29.9:
            print("Obesidad")
            ideal_weight = pIdeal(BMI_NORMAL, h)
            peso_ideal.append(ideal_weight)
            print(f"\nTu peso ideal es: {ideal_weight:.2f} kg")
            o += 1
    
    porc_baj = (100 * b) / users
    porc_nor = (100 * n) / users
    porc_sob = (100 * s) / users
    porc_ob = (100 * o) / users
    
    print(f"\nPorcentaje de personas en Bajo peso: {porc_baj:.1f}%")
    print(f"Porcentaje de personas en Peso Normal: {porc_nor:.1f}%")
    print(f"Porcentaje de personas en Sobrepeso: {porc_sob:.1f}%")
    print(f"Porcentaje de personas en Obesidad: {porc_ob:.1f}%")

if __name__ == "__main__":
    main()
